rmw_connext_shared_cpp/src/node_names.cpp

```diff
@@ -67,15 +71,29 @@ get_node_names(
     return RMW_RET_BAD_ALLOC;
   }
 
-
   for (auto i = 1; i < length; ++i) {
     DDS::ParticipantBuiltinTopicData pbtd;
     auto dds_ret = participant->get_discovered_participant_data(pbtd, handles[i - 1]);
-    const char * name = pbtd.participant_name.name;
-    if (!name || dds_ret != DDS_RETCODE_OK) {
-      name = "(no name)";
+    std::string name;
+    if (DDS_RETCODE_OK == dds_ret) {
+      auto data = static_cast<unsigned char *>(pbtd.user_data.value.get_contiguous_buffer());
+      std::vector<uint8_t> kv(data, data + pbtd.user_data.value.length());
+      auto map = rmw::impl::cpp::parse_key_value(kv);
+      auto found = map.find("name");
+      if (found != map.end()) {
+        name = std::string(found->second.begin(), found->second.end());
+      }
+      if (name.empty()) {
+        // use participant name if no name was found in the user data
+        name = pbtd.participant_name.name;
```

Comment:
The participant name might be null (that seems to be the case for whatever additional participants opensplice is spawning). If we just store the participant name without checking, this causes a segfault at the `if (name.empty())` check on the `std::string` [below](https://github.com/ros2/rmw_connext/pull/276/files#diff-bccfea1954134a50cf75b01ae95188e3R91)

Reply:
That makes sense. While the participant name is a `std::string` in other implementations it is a `char *` in Connext. I checked for `NULL` in ae30090bcfacb88081f4031c8965493e654347fd before assigning it to a `std::string`.